packages/ns-api/files/ns.dpireport

```diff
@@ -26,94 +28,158 @@ def list_days():
-        client = client_f.removeprefix(f'/var/run/dpireport/{year}/{month}/{day}/')
-        cdetails = _details(year, month, day, client)
-        ret["clients"][client] = cdetails["total"]
-        ret["names"][client] = cdetails["name"]
-        for hour in cdetails["hours"]:
-            try:
-                ret["hours"][hour] += cdetails["hours"][hour]["total"]
-            except:
-                pass
-        ret["total"] += cdetails["total"] 
-
-        for key in ("protocol", "host", "application"):
-            for el in cdetails[key]:
-                if el not in ret[key]:
-                    ret[key][el] = 0
-                ret[key][el] += cdetails[key][el]
-    try:
-        tmp_c = sorted(ret['clients'].items(), key=lambda x:x[1], reverse=True)
-    except:
-        tmp_c = list()
-    try:
-        tmp_h = sorted(ret['hours'].items())
-    except:
-        tmp_h = list()
-
-    ret['clients'] = tmp_c
-    ret['hours'] = tmp_h
-
-    for key in ("protocol", "host", "application"):
-        try:
-            tmp = sorted(ret[key].items(), key=lambda x:x[1], reverse=True)
-        except:
-            tmp = list()
-        ret[key] = tmp[0:10]
-
-    return ret
-
-def _details(year, month, day, client):
-    hours = dict()
-    if not client:
-        return ret
-    for i in range(24):
-        hours[f'{i:02}'] = dict()
-    ret = {"hours": hours, "total": 0, "name": reverse_dns(client), "protocol" : {}, "host": {}, "application": {}}
-    ddir = f'/var/run/dpireport/{year}/{month}/{day}/{client}/'
-    for hour_f in glob.glob(f'{ddir}??.json'):
-        hour = hour_f.removesuffix(".json").removeprefix(ddir)
-        with open(hour_f, 'r') as fp:
-            ret["hours"][hour] = json.load(fp)
-        for key in ("protocol", "host", "application"):
-            for el in ret["hours"][hour][key]:
-                if el not in ret[key]:
-                    ret[key][el] = 0
-                ret[key][el] += ret["hours"][hour][key][el]
-        ret["total"] += ret["hours"][hour]["total"]
-    return ret
+
+def _load_data(year, month, day):
+    search_path = f'{BASE_PATH}/{year}/{month}/{day}/'
+    clients = dict()
+    for client_file in glob.glob(f'{search_path}/*'):
+        client_name = client_file.removeprefix(search_path)
+        client_data = dict()
+        client_hourly = glob.glob(f'{client_file}/*.json')
+        for data_file in client_hourly:
+            time = data_file.removeprefix(f'{client_file}/').removesuffix('.json')
+            client_data[time] = _extract_data(data_file)
+        clients[client_name] = client_data
+
+    return clients
+
+
+def _extract_data(dpi_file: str):
+    with open(dpi_file, 'r') as file:
+        json_file = json.load(file)
+        data = {
+            "total": json_file["total"],
+            "protocol": json_file["protocol"],
+            "application": json_file["application"],
+            "host": json_file["host"]
+        }
+        return data
```

Comment:
The format inside the file is already the one describe, why not just returning it?
```suggestion
        return json.load(file)
```

Also it's better to put a try/except and return an empty object in case of error

Reply:
Fixed, careful for the next review, cause I've actually backported the `summary` function to avoid breaking changes.